templates/server/puppetserver/conf.d/auth.conf.erb

```diff
@@ -24,7 +24,11 @@ authorization: {
                 type: regex
                 method: post
             }
+<%- if @server_trusted_agents.empty? -%>
             deny: "*"
+<%- else -%>
+            allow: <%= @server_trusted_agents %>
```

Comment:
I also would have liked to introduce a new parameter so that certificate extensions could have been used as well as just a list of certnames. irb instead of the old way (removed in https://github.com/theforeman/puppet-puppet/pull/615) makes it more difficult though.  If I just append a hash to the `server_trusted_agents` array the outputted string isn't valid hocon.  Valid configuration would look a bit like...

```
allow: [
        "cert1",
        "cert2",
        {
            extensions: {
                pp_authorization: "catalog"
            }
        }
    ]
```
I might try to figure out a nice way of generating this in a separate PR.

Reply:
FYI, PE rule has additional restriction for v4 catalog `query-params: {}`, I presume for security reasons

here is the code snippet:

```
        {
            "allow": [
                "puppet.chepkov.lan",
                {
                    "rbac": {
                        "permission": "puppetserver:compile_catalog:*"
                    }
                }
            ],
            "match-request": {
                "method": "post",
                "path": "^/puppet/v4/catalog/?$",
                "query-params": {},
                "type": "regex"
            },
            "name": "puppetlabs v4 catalog",
            "sort-order": 500
        }
    ,
```

Reply:
It wouldn't hurt.  Don't think it does much either though.  The body of the post is json containing all the input data.  I think any query parameters would just be ignored? https://puppet.com/docs/puppetserver/latest/puppet-api/v4/catalog.html